pkg/vcs/linux.go

```diff
@@ -211,32 +218,63 @@ func (ctx *linux) addMaintainers(com *Commit) {
 	if len(com.CC) > 2 {
 		return
 	}
-	list := ctx.getMaintainers(com.Hash, false)
-	if len(list) < 3 {
-		list = ctx.getMaintainers(com.Hash, true)
+	mtrs := ctx.getMaintainers(com.Hash, false)
+	if len(mtrs) < 3 {
+		mtrs = ctx.getMaintainers(com.Hash, true)
 	}
-	com.CC = email.MergeEmailLists(com.CC, list)
+	com.Recipients = mtrs
 }
 
-func (ctx *linux) getMaintainers(hash string, blame bool) []string {
+func (ctx *linux) getMaintainers(hash string, blame bool) []RecipientInfo {
 	// See #1441 re --git-min-percent.
 	args := "git show " + hash + " | " +
 		filepath.FromSlash("scripts/get_maintainer.pl") +
-		" --no-n --no-rolestats --git-min-percent=20"
+		" --git-min-percent=20"
 	if blame {
 		args += " --git-blame"
 	}
 	output, err := osutil.RunCmd(time.Minute, ctx.git.dir, "bash", "-c", args)
 	if err != nil {
 		return nil
 	}
-	var list []string
-	for _, line := range strings.Split(string(output), "\n") {
-		addr, err := mail.ParseAddress(line)
+	lines := strings.Split(string(output), "\n")
+	return ParseMaintainers(lines)
+}
+
+func ParseMaintainers(lines []string) []RecipientInfo {
+	if len(lines) < 1 {
+		return []RecipientInfo{}
+	}
+	reType := regexp.MustCompile(`\(.*?\)`)
+	var mtrs []RecipientInfo
+	listFlag := false // check if there are lists besides linux-kernel
+	for _, line := range lines {
+		mtr := RecipientInfo{}
+		reType := reType.FindString(line)
+		addr, err := mail.ParseAddress(strings.Replace(line, reType, "", 1))
```

Comment:
Will it correctly handle names like these?
```
M:	James (Qian) Wang <james.qian.wang@arm.com>
M:	Wei Hu(Xavier) <xavier.huwei@huawei.com>
M:	"Richard Russon (FlatCap)" <ldm@flatcap.org>
M:	David (ChunMing) Zhou <David1.Zhou@amd.com>
M:	Juri Lelli <juri.lelli@redhat.com> (SCHED_DEADLINE)
M:	Vincent Guittot <vincent.guittot@linaro.org> (SCHED_NORMAL)
```

Please add tests with names like these.

Reply:
Regarding the last two cases: the line in MAINTAINERS includes the part in parentheses, which presumably is meant to be the part of the subsystem the person specializes in.  However, `get_maintainer.pl` discards this part and doesn't show it.  So I don't think we need to handle those two cases.

We do need to handle other cases where the name contains parentheses, and cases where the name is quoted.

Reply:
And looking at the code more closely, names containing parentheses aren't handled correctly since `reType` will match them.  The regex needs to be written correctly to match just the role.